### PR TITLE
Fix date picker

### LINKED ISSUE #14002: 日期选择框 DatePicker，选择完日期后直接得到焦点，没有考虑是否传了open=true参数，导致点击页面任何地方都回滚到输入处
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant-design.gitee.io/components/date-picker-cn/#header](https://ant-design.gitee.io/components/date-picker-cn/#header)

### Steps to reproduce
例如这样用：
``` javascript
<MonthPicker open={true} />
```

[#12475](https://github.com/ant-design/ant-design/issues/12475),感觉这个bug修的比较仓促：[https://github.com/ant-design/ant-design/commit/d81364c7a9cf90511f71603e801f09a9f6f3a753#diff-34c1c9d6622f2b32f87f4de9a64f099b](https://github.com/ant-design/ant-design/commit/d81364c7a9cf90511f71603e801f09a9f6f3a753#diff-34c1c9d6622f2b32f87f4de9a64f099b)

### What is expected?
如果props中有open并且等于true时，不自动获取焦点

### What is actually happening?
不管open是什么值都直接获取焦点

| Environment | Info |
|---|---|
| antd | 3.11.6 |
| React | 16.4.2  |
| System | mac os |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by Yangzhedi**
fix[#14002](https://github.com/ant-design/ant-design/issues/14002)

### This is a ...

- [ ] New feature
- [x] Bug fix

### What's the background?
  
> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.
  
### API Realization (Optional if not new feature)
  
> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.
  
### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog provided

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/date-picker/createPicker.tsx [hunk 1]
```diff
@@ -68,6 +68,12 @@ export default function createPicker(TheCalendar: React.ComponentClass): any {
       };
     }
 
+    componentDidUpdate(prevProps: PickerProps, prevState: PickerState) {
+      if (!('open' in prevProps) && prevState.open) {
```
Comment (afc163) [inline components/date-picker/createPicker.tsx]: should be `!('open' in this.props)`

Comment (afc163) [reply components/date-picker/createPicker.tsx]: `!('open' in this.props) && !prevState.open && !this.state.open`

**Comment by afc163** [comment]
Could you trying to add some test cases?

File: components/date-picker/createPicker.tsx [hunk 2]
```diff
@@ -68,6 +68,12 @@ export default function createPicker(TheCalendar: React.ComponentClass): any {
       };
     }
 
+    componentDidUpdate(prevProps: PickerProps) {
+      if ('open' in prevProps && prevProps.open) {
```
Comment (ztplz) [inline components/date-picker/createPicker.tsx]: `if (prevProps.open)`. Just doing this is enough

**Review by ztplz — APPROVED** [review]

File: components/date-picker/createPicker.tsx (hunk 2 above)
Comment (Yangzhedi) [reply components/date-picker/createPicker.tsx]: it had updated

**Comment by Yangzhedi** [comment]
> Could you try to add some test cases?

let me have a try

**Pull request merged**